src/util/ThreadMemberFlags.js

```diff
@@ -0,0 +1,30 @@
+'use strict';
+
+const BitField = require('./BitField');
+
+/**
+ * Data structure that makes it easy to interact with a {@link ThreadMember#flags} bitfield.
+ * @extends {BitField}
+ */
+class ThreadMemberFlags extends BitField {}
+
+/**
+ * @name ThreadMemberFlags
+ * @kind constructor
+ * @memberof ThreadMemberFlags
+ * @param {BitFieldResolvable} [bits=0] Bit(s) to read from
+ */
+
+/**
+ * Bitfield of the packed bits
+ * @type {number}
+ * @name ThreadMemberFlags#bitfield
+ */
+
+/**
+ * Numeric thread member flags. All available properties:
```

Comment:
There is at least 1 flag currently, I just don't know what it is and haven't figured it out. I would like to include before this gets merged though, so if anyone can figure it out (or maybe AJ can tell us).

Reply:
The flags are: ```
export const ThreadMemberFlags = {
  HAS_INTERACTED: 1 << 0,
  ALL_MESSAGES: 1 << 1,
  ONLY_MENTIONS: 1 << 2,
  NO_MESSAGES: 1 << 3,
};
```

But they aren't documented in the official docs, and I don't plan to document them either. Mainly because I don't think any of them are really relevant for bots right now, they're user notification settings really, so there's no use for it. This is more preserving the option for us to add useful booleans to the thread member in the future.

Reply:
Would you suggest not including the flags property in the thread member object then?

Reply:
That's hard for me to answer, I don't know your audience and user base, it's definitely a potential option though, I wouldn't say no to it

Reply:
If they aren't documented officially then I'm happy for them not to be included here either 👍

Reply:
Overall we try to cover the public API (so usually only documented stuff), and don't allow user bots in any way shape or form.